src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/Apprenticeship.cs

```diff
@@ -36,12 +39,43 @@ public ApprenticeshipStatus Status
             }
         }
 
-        
+        public DateTime? CompletionDate { get; set; }
 
         public Apprenticeship()
         {
             DataLockStatus = new List<DataLockStatus>();
             PriceHistory = new List<PriceHistory>();
         }
+
+        public virtual void Complete(DateTime completionDate)
+        {
+            if (Status != ApprenticeshipStatus.Live)
```

Comment:
Why are we using the ApprenticeshipStatus property - in fact I'm not sure why that's here at all, it should not be in this layer :-( ? We have a PaymentStatus property for this.

Reply:
The technical documentation stated

```Enforces the following invariants:
The Apprenticeship must be Live - not Waiting to Start, Paused, Stopped, or Complete
The completionDate must be later than the start of the Apprenticeship```

So it seemed logical to use the ApprenticeshipStatus to determine this. We can change it to use PaymentStatus, but surely this is then duplicating the logic.

Reply:
I think I just don't like _where_ that Status field sits tbh, that big switch statement in the domain model. But it's not part of this ticket, so make use of it (better than PaymentStatus because of the need to differentiate Live from Waiting)